data/campaigns/fri04.wmf/scripting/mission_thread.lua

```diff
@@ -0,0 +1,173 @@
+local seen_reebaud = false
+local seen_amazons = false
+
+local o_defeat_amz = nil
+local o_defeat_emp = nil
+local amazons_defeated_by_reebaud = false
+
+function hurry_up()
+   while true do
+      sleep(10000)
+      if seen_reebaud or p4.defeated then return end
+      if #p2:get_buildings("frisians_headquarters") < 1 then
+         campaign_message_box(reebaud_in_danger)
+         return
+      end
+   end
+end
+
+function warn_empire_expansion()
+   while (
+         #p3:get_buildings("empire_blockhouse") +
+         #p3:get_buildings("empire_sentry") +
+         #p3:get_buildings("empire_outpost") +
+         #p3:get_buildings("empire_barrier") +
+         #p3:get_buildings("empire_tower") +
+         #p3:get_buildings("empire_fortress") +
+         #p3:get_buildings("empire_castle")) < 4 do
+      sleep(100000)
+      if seen_reebaud then return end
+   end
+   campaign_message_box(legate_expands)
+end
+
+function see_amazons(field)
+   if amazons_defeated_by_reebaud then return end
```

Comment:
In my last test game they were defeated by the empire before I could meet them or Reebaud. in this case the message boxes didn't make sense. So we should return here if they are defeated in any case. (we might trigger a message box where Claus is wondering about the strange buildings or remains, if he has not met Rebaud)

Reply:
p3 and p4 were never meant to fight each other. Changed that.